integration_test/integration_test.go

```diff
@@ -298,6 +372,621 @@ func (t *WASMIntegrationTestSuite) Test04_ContractExecution() {
+			stakingtype.NewDelegationResp(
+				t.ZeroRewardValidatorWallet1.ByteAddress,
+				t.ZeroRewardValidatorWallet1.ByteAddress.Bytes(),
+				sdktypes.NewDecFromInt(amt),
+				sdktypes.Coin{
+					Denom:  "axpla",
+					Amount: amt,
+				},
+			),
+		}
+
+		if assert.Equal(t.T(), expected, delegatedList) {
+			fmt.Println("Only one delegator exists. Check OK")
+		} else {
+			fmt.Println("Something wrong in the module")
+			t.T().Fail()
+		}
+	}
+
+	delegationAmt := sdktypes.NewInt(100000000000000)
+	{
+		fmt.Println("Try delegation and should fail...")
+
+		coin := &sdktypes.Coin{
+			Denom:  "axpla",
+			Amount: delegationAmt,
+		}
+
+		delegationMsg := stakingtype.NewMsgDelegate(
+			t.UserWallet1.ByteAddress,
+			t.ZeroRewardValidatorWallet1.ByteAddress.Bytes(),
+			*coin,
+		)
+
+		feeAmt := sdktypes.NewDec(xplaGeneralGasLimit).Mul(sdktypes.MustNewDecFromStr(xplaGasPrice))
+		fee := sdktypes.Coin{
+			Denom:  "axpla",
+			Amount: feeAmt.Ceil().RoundInt(),
+		}
+
+		txhash, err := t.UserWallet1.SendTx(ChainID, delegationMsg, fee, xplaGeneralGasLimit, false)
+		if assert.Error(t.T(), err) && assert.Equal(t.T(), txhash, "") {
+			fmt.Println("Expected failure is occurred.")
+		} else {
+			fmt.Println("Tx sent. Test fail")
+			t.T().Fail()
+		}
+	}
+
+	{
+		fmt.Println("Try redelegation and should fail...")
+
+		coin := &sdktypes.Coin{
+			Denom:  "axpla",
+			Amount: delegationAmt,
+		}
+
+		redelegationMsg := stakingtype.NewMsgBeginRedelegate(
+			t.UserWallet1.ByteAddress,
+			t.ValidatorWallet1.ByteAddress.Bytes(),
+			t.ZeroRewardValidatorWallet1.ByteAddress.Bytes(),
+			*coin,
+		)
+
+		feeAmt := sdktypes.NewDec(xplaGeneralGasLimit).Mul(sdktypes.MustNewDecFromStr(xplaGasPrice))
+		fee := sdktypes.Coin{
+			Denom:  "axpla",
+			Amount: feeAmt.Ceil().RoundInt(),
+		}
+
+		txhash, err := t.UserWallet1.SendTx(ChainID, redelegationMsg, fee, xplaGeneralGasLimit, false)
+		if assert.Error(t.T(), err) && assert.Equal(t.T(), txhash, "") {
+			fmt.Println("Expected failure is occurred.")
+		} else {
+			fmt.Println("Tx sent. Test fail")
+			t.T().Fail()
+		}
+	}
+
+	{
+		fmt.Println("Try undelegation and should fail...")
+
+		coin := &sdktypes.Coin{
+			Denom:  "axpla",
+			Amount: delegationAmt,
+		}
+
+		redelegationMsg := stakingtype.NewMsgUndelegate(
+			t.ZeroRewardValidatorWallet1.ByteAddress.Bytes(),
+			t.ZeroRewardValidatorWallet1.ByteAddress.Bytes(),
+			*coin,
+		)
+
+		feeAmt := sdktypes.NewDec(xplaGeneralGasLimit).Mul(sdktypes.MustNewDecFromStr(xplaGasPrice))
+		fee := sdktypes.Coin{
+			Denom:  "axpla",
+			Amount: feeAmt.Ceil().RoundInt(),
+		}
+
+		txhash, err := t.ZeroRewardValidatorWallet1.SendTx(ChainID, redelegationMsg, fee, xplaGeneralGasLimit, false)
+		if assert.Error(t.T(), err) && assert.Equal(t.T(), txhash, "") {
+			fmt.Println("Expected failure is occurred.")
+		} else {
+			fmt.Println("Tx sent. Test fail")
+			t.T().Fail()
+		}
+	}
+
+	{
+		fmt.Println("Preparing proposal to remove a zero reward validator")
```

Comment:
It seems necessary to check whether the "Zero Reward Validator" operates normally in both cases when it belongs to the active set and when it does not belong to it.

Reply:
https://github.com/xpladev/xpla/pull/84/commits/c4ef4ccdcf1603bc2fe5e319e93ef804018c224d